packages/babel-core/src/transformation/file/index.js

```diff
@@ -241,7 +240,18 @@ export default class File extends Store {
       return t.memberExpression(runtime, t.identifier(name));
     }
 
-    const ref = getHelper(name);
+    const opts = {};
+    if (deps) {
+      for (const dep in deps) {
+        if (!this.usedHelpers[name]) {
+          throw "Helper dependencies must be added to the file first";
+        } else {
+          opts["babelHelpers." + dep] = this.usedHelpers[name];
```

Comment:
Hey. I completely missed this, I guess since it was working on babel tests I assumed it would be generating the correct code. Apparently not. Anyways, I'll track the issues with this feature separately in https://github.com/babel/babel/pull/6058. Also I don't intend to implement too complex features for this - just enough to make it work, because #5706 already is aimed at providing a superset of the functionality